src/main/scala/firrtl/annotations/Annotation.scala

```diff
@@ -42,6 +42,9 @@ trait Annotation extends Product {
   */
 trait NoTargetAnnotation extends Annotation {
   def update(renames: RenameMap): Seq[NoTargetAnnotation] = Seq(this)
+  override def productArity: Int = 0
+  override def canEqual(that: Any): Boolean = true
```

Comment:
While I think this is fine as-is, would it be better to do?

```suggestion
  override def canEqual(that: Any): Boolean = that match {
    case _: NoTargetAnnotation => true
    case _                     => false
  }
```